engine/engine/content/builtins/scripts/debugger.lua

```diff
@@ -63,28 +63,30 @@ local function remove_breakpoint(line, file)
   end
 end
 
+local file_cache = {}
 -- This hook fires only on Lua functions calls and checks if this function has lines
 -- we have breakpoints on
-local function lightweight_hook()
-  local caller = debug.getinfo(2,"S")
-  local file = mobdebug.get_filename(caller.source)
+local function lightweight_hook(source, lastlinedefined, co)
+  local file = file_cache[source]
+  if not file then 
+    file = mobdebug.get_filename(source)
+    file_cache[source] = file
+  end
   mobdebug.get_server_data(file)
   local brk = breakpoints[file]
-  local co = coroutine.running()
```

Comment:
here I just avoid extra Lua<->Cpp communications + avoid creation of the table  (`debug.getinfo()`) for each Lua function call